include/exadg/compressible_navier_stokes/spatial_discretization/calculators.h

```diff
@@ -176,6 +176,81 @@ class DivergenceCalculator
   unsigned int quad_index;
 };
 
+template<int dim, typename Number>
+class ShearRateCalculator
```

Comment:
What is exactly the difference to the incompressible case? It looks like we have somewhat copied the code in the past for other calculators as well (divergence, vorticity). Since this has been a bad style, we should now work on removing such code redundancies. So the goal should be to define certain calculators in a general module (e.g. `exadg/operators` and living in the namespace `ExaDG`) and use them where possible, i.e. the same implementation for incompressible and compressible flow.

For the present PR I would suggest that we try to figure out whether we can use the same implementation for both compressible and incompressible (or what we have to change so that this is possible). In a follow-up PR we can then apply this to other calculators relevant for both compressible and incompressible flows.

Reply:
There is no difference whatsoever, but since I found the `calculators` defined separately, I added my code accordingly. I will have a look and keep you posted on further possibilities.

Reply:
I moved the 'calculators' (for now this contains only the shear rate) to 'include/operators/navier_stokes_calculators', copying the one from the incompressible Navier-Stokes directory. Changed the remaining files accordingly and tested w. 'poiseuille' examples in ('in'-)'compressible_navier_stokes'.
As you suggest: in a follow-up PR, we can pull over the rest of the calculators, i.e., vorticity, divergence, velocity magnitude and q-criterion.